### PR TITLE
[CosmosDB] Removing PITR commands and adding Services and Graphs commands

### PR DESCRIPTION
**Pull request by MehaKaushik**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CosmosDB

File: src/cosmosdb-preview/azext_cosmosdb_preview/_params.py [hunk 1]
```diff
@@ -5,77 +5,16 @@
...
-    with self.argument_context('cosmosdb restorable-database-account show') as c:
-        c.argument('location', options_list=['--location', '-l'], help="Location", required=False)
-        c.argument('instance_id', options_list=['--instance-id', '-i'], help="InstanceId of the Account", required=False)
-
-    with self.argument_context('cosmosdb restorable-database-account list') as c:
-        c.argument('location', options_list=['--location', '-l'], help="Location", required=False)
-        c.argument('account_name', options_list=['--account-name', '-n'], help="Name of the Account", required=False, id_part=None)
-
     # Restorable Sql Databases
     with self.argument_context('cosmosdb sql restorable-database') as c:
```
Comment (kavskalyan) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_params.py:19]: All the sql, mongo restorable commands can be removed

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 2]
```diff
@@ -7,31 +7,23 @@
...
 from azure.cli.core.commands import CliCommandType
 from azext_cosmosdb_preview._client_factory import (
-    cf_db_accounts,
-    cf_restorable_database_accounts,
     cf_restorable_sql_databases,
     cf_restorable_sql_containers,
     cf_restorable_sql_resources,
     cf_restorable_mongodb_databases,
     cf_restorable_mongodb_collections,
     cf_restorable_mongodb_resources,
```
Comment (kavskalyan) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:15]:
```suggestion
```

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 3]
```diff
@@ -7,31 +7,23 @@
...
-        operations_tmpl='azext_cosmosdb_preview.vendored_sdks.azure_mgmt_cosmosdb.operations#DatabaseAccountsOperations.{}',
-        client_factory=cf_db_accounts)
-
-    cosmosdb_restorable_database_accounts_sdk = CliCommandType(
-        operations_tmpl='azext_cosmosdb_preview.vendored_sdks.azure_mgmt_cosmosdb.operations#RestorableDatabaseAccountsOperations.{}',
-        client_factory=cf_restorable_database_accounts)
-
     cosmosdb_restorable_sql_databases_sdk = CliCommandType(
         operations_tmpl='azext_cosmosdb_preview.vendored_sdks.azure_mgmt_cosmosdb.operations#RestorableSqlDatabasesOperations.{}',
         client_factory=cf_restorable_sql_databases)
```
Comment (kavskalyan) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:29]:
```suggestion
```

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 4]
```diff
@@ -64,17 +64,6 @@ def load_command_table(self, _):
...
-
-    with self.command_group('cosmosdb', cosmosdb_sdk, client_factory=cf_db_accounts) as g:
-        g.show_command('show', 'get')
-        g.custom_command('restore', 'cli_cosmosdb_restore', is_preview=True)
-        g.custom_command('create', 'cli_cosmosdb_create')
-        g.custom_command('update', 'cli_cosmosdb_update')
-        g.custom_command('list', 'cli_cosmosdb_list')
-
     with self.command_group('cosmosdb sql restorable-database', cosmosdb_restorable_sql_databases_sdk, client_factory=cf_restorable_sql_databases, is_preview=True) as g:
         g.command('list', 'list')
```
Comment (kavskalyan) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:69]:
```suggestion
```

**Review by kavskalyan — APPROVED** [review]

**Comment by MehaKaushik** [comment]
@evelyn-ys do you know what is the reason of the failure in this https://dev.azure.com/azure-sdk/public/_build/results?buildId=1111224&view=logs&j=ccdac6d3-465a-5ada-8819-d5b4a6b3a631&t=ffdbba7d-e523-5908-c7bf-0703bafb81b0

**Comment by evelyn-ys** [comment]
> @evelyn-ys do you know what is the reason of the failure in this https://dev.azure.com/azure-sdk/public/_build/results?buildId=1111224&view=logs&j=ccdac6d3-465a-5ada-8819-d5b4a6b3a631&t=ffdbba7d-e523-5908-c7bf-0703bafb81b0

It's because no help entry was found for command group `cosmosdb graph` and `cosmosdb service`

File: src/cosmosdb-preview/azext_cosmosdb_preview/_help.py [hunk 5]
```diff
@@ -285,3 +145,66 @@
...
+"""
+
+helps['cosmosdb graph delete'] = """
+type: command
+short-summary: Delete the given cosmosdb graph resource.
+"""
+
+helps['cosmosdb graph exists'] = """
+type: command
+short-summary: Returns if the given cosmosdb graph resource exist.
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_help.py:209]:
```suggestion
short-summary: Return if the given cosmosdb graph resource exist.
```

File: src/cosmosdb-preview/azext_cosmosdb_preview/_params.py [hunk 6]
```diff
@@ -175,3 +78,19 @@ def load_arguments(self, _):
...
+
+    # Graph
+    with self.argument_context('cosmosdb graph') as c:
+        c.argument('account_name', completer=None, options_list=['--account-name', '-a'], help='Name of the Cosmos DB database account.', id_part=None)
+        c.argument('resource_group_name', completer=None, options_list=['--resource-group-name', '-g'], help='Name of the resource group of the database account.', id_part=None)
+        c.argument('graph_name', options_list=['--name', '-n'], help="Graph name")
+
+    # Services
+    for scope in ['cosmosdb service']:
+        with self.argument_context(scope) as c:
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_params.py:90]:
```suggestion
        with self.argument_context('cosmosdb service') as c:
```

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 7]
```diff
@@ -107,3 +48,17 @@ def load_command_table(self, _):
...
         g.show_command('show', 'get')
         g.command('delete', 'begin_delete', confirmation=True, supports_no_wait=True)
+
+    with self.command_group('cosmosdb graph', cosmosdb_graph_resources_sdk, client_factory=cf_graph_resources, is_preview=True) as g:
+        g.custom_command('create', 'cli_cosmosdb_graph_create', supports_no_wait=True)
+        g.custom_command('exists', 'cli_cosmosdb_graph_exists', supports_no_wait=True)
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:54]: Why should `exists` command support `--no-wait`? That's kind of wired ><

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 8]
```diff
@@ -107,3 +48,17 @@ def load_command_table(self, _):
         g.command('list', 'list')
         g.show_command('show', 'get')
         g.command('delete', 'begin_delete', confirmation=True, supports_no_wait=True)
+
+    with self.command_group('cosmosdb graph', cosmosdb_graph_resources_sdk, client_factory=cf_graph_resources, is_preview=True) as g:
+        g.custom_command('create', 'cli_cosmosdb_graph_create', supports_no_wait=True)
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:53]: Just to confirm, there's no `update` command for `cosmosdb graph`, right?

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py [hunk 9]
```diff
@@ -604,6 +218,63 @@ def cli_cosmosdb_managed_cassandra_datacenter_update(client, resource_group_name
...
+def cli_cosmosdb_graph_exists(client, account_name, resource_group_name, graph_name):
+    try:
+        client.get_graph(resource_group_name, account_name, graph_name)
+    except HttpResponseError as ex:
+        return _handle_exists_exception(ex)
+
+    return True
+
+
+def cli_cosmosdb_service_exists(client, account_name, resource_group_name, service_name):
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:241]: I didn't see `cosmosdb service exists` command. Is this command missing or is this function extra?

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py [hunk 10]
```diff
@@ -604,6 +218,63 @@ def cli_cosmosdb_managed_cassandra_datacenter_update(client, resource_group_name
...
+    return True
+
+
+def cli_cosmosdb_service_create(client,
+                                account_name,
+                                resource_group_name,
+                                service_kind,
+                                service_name,
+                                instance_count,
+                                instance_size):
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:256]: Are all these parameters required? Do they have default values?

**Review by evelyn-ys — COMMENTED** [review]
Don't forget to update `HISTORY.rst` and upgrade the version in `setup.py` to release a new version.

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py (hunk 8 above)
Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:53]: yes

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py (hunk 10 above)
Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:256]: I can add default values to instance_Count and instance_size, others cannot have defaults

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py (hunk 9 above)
Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:241]: will remove the function

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py (hunk 7 above)
Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:54]: removed

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py (hunk 10 above)
Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/custom.py]: Added for create scenario, not required for update scenario

File: src/cosmosdb-preview/azext_cosmosdb_preview/_help.py [hunk 11]
```diff
@@ -285,3 +145,76 @@
...
+type: group
+short-summary: Commands to perform operations on Graph resources.
+"""
+
+helps['cosmosdb graph create'] = """
+type: command
+short-summary: Create a cosmosdb graph resource.
+"""
+
+helps['cosmosdb graph update'] = """
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_help.py:202]: This command doesn't exist

Comment (MehaKaushik) [reply src/cosmosdb-preview/azext_cosmosdb_preview/_help.py]: removed, thanks for pointing out

**Comment by MehaKaushik** [comment]
@evelyn-ys please merge if it looks good to you. TIA

**Comment by evelyn-ys** [comment]
> @evelyn-ys please merge if it looks good to you. TIA

Can you pls address the [linter failure](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1112019&view=logs&j=ccdac6d3-465a-5ada-8819-d5b4a6b3a631&t=ffdbba7d-e523-5908-c7bf-0703bafb81b0&l=479)?
```


	"az cosmosdb service create --resource-group MyResourceGroup --account-name MyAccount --service-name "graphApiCompute" --service-kind "GraphApiCompute" --count 1 --size "Cosmos.D4s"" is not a valid command.
	the following arguments are required: --kind/-k, --name/-n


    Help-Entry: `cosmosdb service delete` - 
	There is a violation:
	"az cosmosdb service delete --resource-group MyResourceGroup --account-name MyAccount --service-name "sqlDedicatedGateway"" is not a valid command.
	the following arguments are required: --name/-n.


    Help-Entry: `cosmosdb service update` - 
	There are 2 violations:
	"az cosmosdb service update --resource-group MyResourceGroup --account-name MyAccount --service-name "sqlDedicatedGateway" --service-kind "SqlDedicatedGateway" --count 3" is not a valid command.
	the following arguments are required: --name/-n, --kind/-k

	"az cosmosdb service update --resource-group MyResourceGroup --account-name MyAccount --service-name "graphApiCompute" --service-kind "GraphApiCompute" --count 1" is not a valid command.
	the following arguments are required: --name/-n, --kind/-k

```

**Pull request merged**